### ISSUE TITLE
Needs Support for 10.11.2

### ISSUE THREAD

**Issue by rasmi** [Floris497/mac-pixel-clock-patch#18]
Here are the checksums I have for OS X 10.11.2, which was released today:
```
$ md5 -q /System/Library/Frameworks/IOKit.framework/Versions/A/IOKit
a7afb2dd9df1e4c48f12b4b52f7da212
$ md5 -q /System/Library/Extensions/NVDAGK100Hal.kext/Contents/MacOS/NVDAGK100Hal
bb87a13eaabefcafa9e82fad03365aa4
$ md5 -q /System/Library/Extensions/AMDSupport.kext/Contents/MacOS/AMDSupport
018b02afbcaeca2883b316076bae2700
```

**Comment by legotheboss** [comment]
I tried the commands for the previous 10.11.1 and it executed perfectly fine. Just manually run the commands for 10.11.1 if this is important for you.

**Comment by rasmi** [comment]
@legotheboss, thanks for the tip. I can confirm that it worked fine for me, too. Here's my checksum for the patched IOKit:
```
md5 -q /System/Library/Frameworks/IOKit.framework/Versions/A/IOKit
a9435069e490b38e807a0d9b2eb6e76c
```
And I went ahead and created a [pull request](https://github.com/Floris497/mac-pixel-clock-patch/pull/19) for the IOKit patch. Someone else will have to take care of the Nvidia and AMD patches -- I can't test them on my machine, but I suspect the changes are similarly minimal.

**Comment by Nerokrox** [comment]
i tried your IOKit patch. It says:
Detected unpatched IOKit on 10.11.2, patching.

Can't rename /System/Library/Frameworks/IOKit.framework/Versions/A/IOKit to /System/Library/Frameworks/IOKit.framework/Versions/A/IOKit.bak: Operation not permitted, skipping file.
/System/Library/Frameworks/IOKit.framework/Versions/A/IOKit: replacing existing signature
/System/Library/Frameworks/IOKit.framework/Versions/A/IOKit: Operation not permitted

Can you help me?

**Comment by Floris497** [comment]
System integrity protection

**Comment by pos005** [comment]
Can confirm this is working on my hackintosh with a Titan X. 4k 60hz.
Thanks!

**Comment by gaocuo** [comment]
FYI

iokit_md5_10_11_2=a7afb2dd9df1e4c48f12b4b52f7da212

iokit_md5_10_11_2_patched=5718d75dc7390e3209bb7f8092e7d05b

and confirm it working on my i5-4590.

**Comment by Floris497** [comment]
@pos005 is this patch needed on a titan x?? I don't need it on my 670

**Comment by pos005** [comment]
@Floris497 yes it is. I use my Titan x to a 4K tv via hdmi 2.0. Without this patch it does not output any resolution at 60hz. With the patch everything works normally.

**Comment by Floris497** [comment]
Oh good to know!

**Comment by rwkyyy** [comment]
Worked for me on Air early <blockquote>Detected unpatched IOKit on 10.11.2, patching.

Password:

/System/Library/Frameworks/IOKit.framework/Versions/A/IOKit: replacing existing signature</blockquote>

**Comment by rasmi** [comment]
Great! @Floris497, should this issue be closed, or is there more work that needs to be done (specifically for the Nvidia patch)?

**Comment by Floris497** [comment]
@rwkyy why is this patch needed on a 2015 machine? I tought it was only needed for older hardware..

**Comment by rwkyyy** [comment]
@Floris497 because I got an external display with resolution up to 2560 x 1080 @60hz and OS X allows 1920 x 1080 on full HD (using a display port adapter to hdmi).

**Comment by Floris497** [comment]
That is weird.. You should be able to add that resolution using switchresx without this patch, I think. Also I think 2015 models will output that resolution over HDMI no problem, although native 2560x1440 screens seem to work better than 2560x1080..

**Comment by mutatioveritas** [comment]
"Hi this stopped working on 10.11.2 for me. When I run the commands nothing happens it doesn't provide any output. Tried it with disabling rootless too

Edit: Got it working with the three extra command lines"

I'm in the same boat as Kazom was initially. No response with the 3 provided command lines. Tried with and without SIP enabled. Any help would be great.

**Comment by rwkyyy** [comment]
@Floris497 Searched like 3 days for a solution to display the native resolution, with switchres I only managed to get a virtual resolution, but I still wanted native for the sharp text and stuff.

This patch was the only one that solved my problem.

**Comment by Floris497** [comment]
@rwkyyy good to know
@mutatioveritas does it work now?

**Comment by mutatioveritas** [comment]
Got it to run once and got an error about an xcrun folder being missing. Installed Xcode thinking that might help. Tried it a number of times thereafter and back to the no response.

**Comment by Floris497** [comment]
Place the old file back using the mv command, then redo the patch. The exact command is somwhere in the other issues here or on the master repository.. I'll link it later if i have time

**Comment by mutatioveritas** [comment]
Where does it end up at? Tried looking in the downloads folder as well as the home drive with all files visible. I'll try the mv command when I get a sense of which directory it's in.

**Comment by rasmi** [comment]
@mutatioveritas, the three commands are [here](https://github.com/Floris497/mac-pixel-clock-patch/blob/master/macPixelClockPatcher.command#L323-L326), not the ones posted above:
```
sudo perl -i.bak -pe '$before = qr"\x0f\x85\x92\x03\x00\x00"s;s/$before/\xe9\x7a\x03\x00\x00\x90/g' /System/Library/Frameworks/IOKit.framework/Versions/A/IOKit
sudo touch /System/Library/Extensions
# Now appears to require re-signing, despite not being in CodeResources
sudo codesign -f -s - /System/Library/Frameworks/IOKit.framework/Versions/A/IOKit
```
the backup file is in the same directory that's being modified and should be called `/System/Library/Frameworks/IOKit.framework/Versions/A/IOKit.bak` , but try running the three commands above with SIP disabled.

**Comment by mutatioveritas** [comment]
Thanks, rasmi. Just to clarify, I'm running those commands after having run the initial three commands (e.g., cd ~/Downloads, etc.)? For instance, are the commands you've offered supposed to be run after getting the error that I got, where I'm now left without the initial commands getting any traction? Or, are the commands you've included intended to be run assuming the initial commands were successful?

**Comment by Floris497** [comment]
They should work either way

**Comment by skubic** [comment]
Thanks a lot for your work. U had succsess in many cases with this patch. Now im stucked. 
OS X 10.11.2 with GTX970 Gaming F1 and latest Nvidia Web Drivers. Samsung UHD TV connected to HDMI2. Works without Patch at 3840x2160@30hz. So I tried your patch to enable 60hz. Some Useres were able to get 60hz then. Not me. Do I need to patch the Nvidia Web Driver or is it just the IOKit?

**Comment by mirarty** [comment]
I have similar issue as skubic with GTX680MX. In windows I'm able to get 52Hz but in OS X 10.11.2 only 30Hz. Since I don't have thunderbolt 2 I know the max frequency will be 52Hz but it is not supported in latest driver. Any suggestion how to patch nvidia driver to overclock the pixel clock?

**Comment by rasmi** [comment]
@skubic  and @mirarty, you can try the [following](https://github.com/Floris497/mac-pixel-clock-patch/blob/master/macPixelClockPatcher.command#L364-L365):

```
	sudo perl -i.bak -pe '$oldLimit1 = qr"\xC7\x82\xD0\x00\x00\x00\x88\x84\x02\x00"s;$newLimit1 = "\xC7\x82\xD0\x00\x00\x00\x80\x1A\x06\x00";$oldLimit2 = qr"\xC7\x82\x20\x01\x00\x00\x88\x84\x02\x00"s;$newLimit2 = "\xC7\x82\x20\x01\x00\x00\x80\x1A\x06\x00";s/$oldLimit1/$newLimit1/g;s/$oldLimit2/$newLimit2/g' /System/Library/Extensions/NVDAGK100Hal.kext/Contents/MacOS/NVDAGK100Hal
	sudo touch /System/Library/Extensions
```
It would also be helpful if you could both share your md5 checksums so they can be included in the script (before and after patching).
```
md5 -q /System/Library/Extensions/NVDAGK100Hal.kext/Contents/MacOS/NVDAGK100Hal
```

**Comment by mirarty** [comment]
@rasmi 
Before patching:  bb87a13eaabefcafa9e82fad03365aa4
After patching:  3b3244d597be457326d9c19309f00ff0

I don't see any difference after patching. My current limit is 358.20MHz Any other suggestion what I can try?

**Comment by rasmi** [comment]
@mirarty, did you restart after patching?

**Comment by Floris497** [comment]
if you can give me the version of the driver you are using I'll see if a new patch is needed..
See this one: https://github.com/Floris497/mac-pixel-clock-patch/issues/14 is quite recent.

(I wrote this 12h ago.. But didn't press comment.. 😁)

**Comment by skubic** [comment]
@rasmi 
Before patching: 3b3244d597be457326d9c19309f00ff0
After patching: 

@Floris497 
NVIDIA Web Driver: 346.03.04f01

**Comment by Floris497** [comment]
If the md5 is the same the patch did not work.. I'll try to make a new patch in the comming days. It's nit to much work, basically you change 1 or 2 instances of the value 165Mhz (not sure if in code 165000 or 165000000 the initial patch changed this to 400Mhz maybe for 4k@60Hz it needs to be even heigher if you give this a try remember bytes are reversed (little/big endian IIRC))

**Comment by skubic** [comment]
Thanks for your Supoport Floris497. I can't do it, since I do not nkow wich values to change. Looking forward to your new patch.

**Comment by mirarty** [comment]
@rasmi Yes I did restart after patching
@Floris497  I tried both - default OS X driver and Nvidia web driver as well (346.03.04f01)
@skubic You probably applied patch to already patched pixel clock. That's why you have same md5. You can try to recover .bak, restart system and check md5 again. Then should be same as mine before patching.

**Comment by Devolv** [comment]
I'm having the same issues @Kazom. Nothing happens after I run the commands. I'm on 10.11.2 with a MacBook Pro Mid-2012.  SIP is disabled.

$ md5 -q /System/Library/Frameworks/IOKit.framework/Versions/A/IOKit
4be579207b74e491f8b42e5080d5c396

$ md5 -q /System/Library/Extensions/NVDAGK100Hal.kext/Contents/MacOS/NVDAGK100Hal
ae057feda04e4f0b3c6ef8ba9f75c13e

It looks like my versions are completely different.  Did I overpatch this and screw it up?  How do I restore the backups?

**Comment by rasmi** [comment]
To restore backups:
```
mv /System/Library/Frameworks/IOKit.framework/Versions/A/IOKit.bak /System/Library/Frameworks/IOKit.framework/Versions/A/IOKit
```
and 
```
mv /System/Library/Extensions/NVDAGK100Hal.kext/Contents/MacOS/NVDAGK100Hal.bak /System/Library/Extensions/NVDAGK100Hal.kext/Contents/MacOS/NVDAGK100Hal
```

**Comment by mutatioveritas** [comment]
As an update, still haven't gotten anywhere with the suggested commands. As an aside, is there any reason I couldn't install the drivers on my MacBook Pro, for example, run the commands, and then use a kext installer to get them ported over? As an aside, the machine I'm having issues with is a Hackintosh and the larger issue might be the apparent inability to affect the SIP settings. e.g., I can disable them in recovery mode, reboot, and then check the status in El Capitan and the custom settings are right back where they were prior to the disable command. :-S Any thoughts?

**Comment by mutatioveritas** [comment]
Oh yeah, just got the prompt for 346.03.04f02 web drivers as well. Zoiks!

**Comment by Floris497** [comment]
I'll download the latest web drivers and see if i can patche them 😃 I'll not change the max to 400Mhz but 800Mhz so we are sure te 4k@60 will work

**Comment by mirarty** [comment]
Hopefully you will able to patch them :smile: Anyway thank you for your work!

**Comment by skubic** [comment]
@Floris497 

That sounds fantastic. And it saves money. Club 3D has an active DP > HDMI @ 60Hz introduced adapter. It should cost around 40 €. But I'd rather use HDMI2. Do you accept donations for your work?

**Comment by Floris497** [comment]
this is a working patch for the newest web driver:
```bash
$ sudo perl -i.bak -pe '$oldLimit1 = qr"\xC7\x82\x80\x00\x00\x00\x88\x84\x02\x00"s;$newLimit1 = "\xC7\x82\x80\x00\x00\x00\xC0\x27\x09\x00";$oldLimit2 = qr"\xC7\x82\xF8\x00\x00\x00\x88\x84\x02\x00"s;$newLimit2 = "\xC7\x82\xF8\x00\x00\x00\xC0\x27\x09\x00";s/$oldLimit1/$newLimit1/g;s/$oldLimit2/$newLimit2/g' /System/Library/Extensions/NVDAGK100HalWeb.kext/Contents/MacOS/NVDAGK100HalWeb
$ sudo touch /System/Library/Extensions

```
care to try it before i summit it?
I changed it to 600Mhz should be sufficient for 4k@60Hz

**Comment by skubic** [comment]
Fresh OS X 10.11.2 installaton with latest Nvidia Web Drivers 346.03.0f02 seems to do something.

MD5 Before: 914b2e757cf16e8105a7080792264cc9
MD5 After: 759bb537cfc38c19e46ef15c31f12fcc

But I'm still limited to 30hz. Yes, my TV can do 60hz. On Windows it woks. Hate Windows.

Will now try with SwitchResX

**Comment by Floris497** [comment]
You will need switchresx, also make sure you selected the right drivers, and disabled SIP, SIP needs to stay disabled because it allows unsigned kext files.

**Comment by mirarty** [comment]
Before patching: 914b2e757cf16e8105a7080792264cc9
After patching: 759bb537cfc38c19e46ef15c31f12fcc

I have SIP disabled, I successfully applied patch, I have loaded right web driver, but I can't get more than 358.20MHz. I tried enable Frequency Range in SwitchResX and I set custom Pixel clock frequency to 600MHz, but custom resolution shows Not activated - invalid? Do you guys have any ideas how to get more than 358.20MHz?

**Comment by Floris497** [comment]
Can you look in system information and see which drives are loaded? And what driver is used by the display? Maybe we patched the wrong one, I think we patched the mobile gpu kext.. I'll look what I can find out about my gtx670

**Comment by mirarty** [comment]
I believe you the patched right one. Where I can check what driver is used by the display?
![display](https://cloud.githubusercontent.com/assets/16055929/11873453/7e04f36e-a48f-11e5-89a5-ddbf383e105d.jpg)

**Comment by skubic** [comment]
My Nvidia Web Driver uses NVDAGM100HalWeb. So I patched this file.

![bildschirmfoto 2015-12-17 um 20 35 40](https://cloud.githubusercontent.com/assets/3761217/11880000/ed90df6c-a4fe-11e5-9379-03a7c103a55e.png)

MD5 before: 7f1a015bd7bdb7344c43c64d173999e7
MD5 after: fad34f3b8ca158d8b1e481bd3eef7f10

But still SwitchResX can't activate 60hz.

![bildschirmfoto 2015-12-17 um 20 39 49](https://cloud.githubusercontent.com/assets/3761217/11880014/f8a9678e-a4fe-11e5-9205-a150b45a75e7.png)

**Comment by Floris497** [comment]
You did not change the file, the md5 checksums are from a different file? The driver still has a valid code signature, or is the screenshot from before? Then check if the driver is loaded okay after patching. Also the K patch would not specifically work on the M version, i can do the calculations the day after tomorrow.

**Comment by skubic** [comment]
The first (7f1a015bd7bdb7344c43c64d173999e7) checksum is from before patching. Then I entered:

sudo perl -i.bak -pe '$oldLimit1 = qr"\xC7\x82\x80\x00\x00\x00\x88\x84\x02\x00"s;$newLimit1 = "\xC7\x82\x80\x00\x00\x00\xC0\x27\x09\x00";$oldLimit2 = qr"\xC7\x82\xF8\x00\x00\x00\x88\x84\x02\x00"s;$newLimit2 = "\xC7\x82\xF8\x00\x00\x00\xC0\x27\x09\x00";s/$oldLimit1/$newLimit1/g;s/$oldLimit2/$newLimit2/g' /System/Library/Extensions/NVDAGM100HalWeb.kext/Contents/MacOS/NVDAGM100HalWeb

sudo touch /System/Library/Extensions

Have done a reboot and the checksum changed to the second (fad34f3b8ca158d8b1e481bd3eef7f10).

Asked for the Checksum with:

md5 -q /System/Library/Extensions/NVDAGM100Hal.kext/Contents/MacOS/NVDAGM100Hal

**Comment by Floris497** [comment]
That is not the web driver, but the normal apple driver 😅😅

**Comment by skubic** [comment]
My mistake. Now the "Web" ist attatched. And here is the Screenshot after patching:

![bildschirmfoto 2015-12-17 um 21 38 15](https://cloud.githubusercontent.com/assets/3761217/11881387/df6b2c28-a506-11e5-98f2-12fbc8bc812a.png)

**Comment by Floris497** [comment]
Seems to be good.. I'll look at it the day after tomorrow :-)

**Comment by mirarty** [comment]
Hello Floris, did you had time to take a look what's going on? :smiley:

**Comment by Floris497** [comment]
I have been thinking about it.. But I have no idea where It goes wrong.. Maybe an active adapter is the only solution for now :( what is the heighest framerate you can accomplish with the patch now? Maybe we should also patch the normal apple driver..

**Comment by skubic** [comment]
The highest is 30hz. Have tried nearly every possible Resolution.

**Comment by Floris497** [comment]
Hm.. It's hard for me to discover how to get this working.. I have no GPU that can output HDMI2.0.. I'll see if i find time to digg around a little this year 😁

**Comment by skubic** [comment]
Thank you. I'll give the Club 3D active 60hz DP>HDMI2 a try. As soon as it will be in stores around Europe. Enjoy the Christmas Days.

**Comment by mirarty** [comment]
In my case highest is 3840x2160@40Hz. Which active adapter you recommend?

**Comment by Floris497** [comment]
Over HDMI? That is actually good, I think which means you can output over the 1.4 standart? I'll try if my gtx 670 can do 40Hz over HDMI

**Comment by mirarty** [comment]
@Floris497 I use thunderbolt1 / mDP.

**Comment by Floris497** [comment]
Oh :/ I'll look for something though 😁

**Comment by androcci** [comment]
@Floris497  hello floris. we have the same gnu card. the GTX 670 2gb i believe. i have downloaded your file. on the first set up it asked me for my password after i pasted the command lines. i restarted and switchresx said not activated-invalid?
so i tried doing it again. but when i pasted the command lines, i got NO OUTPUT, nothing to put my password on this time. nothing works...... I'm on el capitan 10.11.2. any help would be much appreciated. here is a scene shot! :)

![screen shot 2015-12-28 at 1 08 29 pm](https://cloud.githubusercontent.com/assets/16467650/12025524/a4327dc6-ad64-11e5-89b1-1e78022a35af.png)
![screen shot 2015-12-28 at 1 09 42 pm](https://cloud.githubusercontent.com/assets/16467650/12025527/a8bfcd08-ad64-11e5-9301-3edc6a56b14a.png)

the second one is proof that the command lines pasted that worked before, don't work now, after i run the program once.

**Comment by mirarty** [comment]
@androcci Do you have disabled SIP?

**Comment by androcci** [comment]
@mirarty yeah!:D :( i went to recovery partition and disabled it then restarted again and i found out about this awesome patch! someone said try switchresx so that is why i disabled SIP. then i downloaded the command file and pasted the command lines on terminal, it worked the first time asked me for my password and all, it only said it found kext but didn't see any nvidia. i restarted and the resolution set was invalid? i dont know what to do! 30Hz is killing me!!! and I'm just lucky i have the same gnu as floris but is not working for me. thanks!!

**Comment by mirarty** [comment]
@androcci What is your MD5?
$ md5 -q /System/Library/Extensions/NVDAGK100HalWeb.kext/Contents/MacOS/NVDAGK100HalWeb

**Comment by androcci** [comment]
i gave up. i talked to floris in another topic. she told me that we have the same computer parts. the problem is that i installed el capitan with a gtx 750 ti then i got my gtx 670 in the mail and i plugged it in. she told me to re install it. i will keep you up date it. i have a good feeling this time it will work. i think the gtx 670 is the GPU to go to. i deleted everything I'm starting from zero again.

**Comment by mirarty** [comment]
Good luck! Keep us posted.

**Comment by Floris497** [comment]
@androcci I pretty much feel like I'm a boy 🙈
I didn't read this before I commend on the other topic, it doesn't matter what card is in when installing.. Can't find a link of the cube 3d adapter, should be the only one on the market now.. I think I linked it before 😁 I'll make the Nvidia patch later, but I'm sure you can find the patch in a different topic, a similar question has been around before..
If you want it the easy way use the 670, if you want the extra performance we have to find out how to get the Ti card to work.

**Comment by androcci** [comment]
@mirarty here is the link to my detailed problem and the md5 you asked about!
https://github.com/vinc3m1/mac-pixel-clock-patch/pull/9#issuecomment-167856264

**Comment by timohour** [comment]
##### It worked great on an H87N board with onboard HD4600 and HDMI output.
From maximum 2048x1152 it now supports 2560x1440. 
But for some reason when I set the resolution @ 1440p it kills the wifi (BCM943225).
Setting the resolution back 2048x1152 makes the wifi work again. I can't' figure out why. 

> ##### Any suggestion?

**Comment by Xenobreaker** [comment]
For what it's worth, I have the Club3D DisplayPort 1.2 to HDMI 2.0 CAC-1070 adapter and a GTX 670.I patched IOKit then I tried the patch Floris497's posted earlier for the latest Nvidia web drivers (346.03.04f02) that also sets the limit to 600mhz which is this:
```
sudo perl -i.bak -pe '$oldLimit1 = qr"\xC7\x82\x80\x00\x00\x00\x88\x84\x02\x00"s;$newLimit1 = "\xC7\x82\x80\x00\x00\x00\xC0\x27\x09\x00";$oldLimit2 = qr"\xC7\x82\xF8\x00\x00\x00\x88\x84\x02\x00"s;$newLimit2 = "\xC7\x82\xF8\x00\x00\x00\xC0\x27\x09\x00";s/$oldLimit1/$newLimit1/g;s/$oldLimit2/$newLimit2/g' /System/Library/Extensions/NVDAGK100HalWeb.kext/Contents/MacOS/NVDAGK100HalWeb
sudo touch /System/Library/Extensions
```
my patched driver now gives an md5 checksum of: 759bb537cfc38c19e46ef15c31f12fcc

Upon testing, I can still only achieve 3840x2160 30hz, there's still no option for 3840x2160 60hz. Out of curiosity, I booted my system up in Windows and I was also still not able to get 60hz as well. Note that I tested out the DisplayPort 1.2 to HDMI 2.0 adapter with a newer card, my GTX 970 and I was able to get 3840x2160 60hz. I'd also like to mention that I already update the adapter to the latest firmware from Club3D. I also created a custom resolution on SwitchResX to match exactly the EDID description provided by my LG 49UB8500 TV.

According to Nvidia's specifications, the GTX 670 should be able to support up to 4096x2160 60hz over its DisplayPort 1.2 port but for some reason, when an adapter is added, it doesn't work up to 60hz even though the same adapter works up to 60hz when used with a GTX 970. I thought it just didn't work on OS X but it didn't work on Windows either.

Any thoughts regarding this? Thanks!

**Comment by Floris497** [comment]
Try patching IOKit aswel, my 670 works great at 4k60Hz on DP.. @androcci was able to get 4k60Hz from HDMI on his 670 in windows.. Interestingly enough.. I will try that next time I have windows working

**Comment by Xenobreaker** [comment]
I patched IOKit first and that was what made the 60hz option appear when I used the adapter with the GTX 970 but when I swapped it out for the GTX 670, there was no option for 60hz. That was when I tried applying the patch you posted for the latest Nvidia web drivers but that didn't work either. 

In @androcci 's case, 4k 60hz through HDMI on Windows works because it's only doing 4:2:0 chroma sub sampling and not at the full 4:4:4. 4k 60hz at 4:2:0 (or 4:2:2 can't remember) will work just fine with HDMI 1.4 and only needs below 300mhz. @androcci can test this by viewing some chroma subsampling test images. When playing full screen 3D games, I doubt you'll notice the difference between 4:4:4 and 4:2:0 but when reading text on a browser, the difference is huge. 

Both the GTX 670 and the GTX 970 use DisplayPort 1.2 so I'm not sure what's causing the discrepancy with the GTX 670 not showing a 3840x2160 60hz option with everything else in the setup being the same.

**Comment by Floris497** [comment]
Oh yeah.. That 4:4:4 4:2:2 forgot about that.. You probably checked it, but are the same drivers loaded with both cards? I'll see if I can find anything else, but I doubt it :/

**Comment by androcci** [comment]
hey guys!!! :). probably you are right i don't know what chroma sub sampling is but i will learn as soon as i get home. i cant just get 4k @60Hz right of the bat with the gtx 670, i need to download the nvidia panel and then download the driver from the panel and then after restarting the machine i get 4k @60hz. i also get it with the HDMI. anyways I'm so over the gtx 670. i ordered the gtx 970 SSC, so its HDMI port works with OS X 4k @60Hz?OOB? i have a Hackintosh tho.

**Comment by androcci** [comment]
doing it at 4:2:2 not bad not bad. I just bought the gtx 970 for $270 such a great deal i couldn't resist. the card is coming january the 25th. i will let you know if that works with my hackintosh!!

**Comment by skubic** [comment]
Just want to give a short Feedback. By doing it as written in my last Post from Dec. 17th it is finaly working. The active Club 3D DP to HDMI2 60Hz. Adapter was the Solution. Without that it wont work for me. Still if I use it, i need to Patch the Nvidia Web File and I need to use SwitchResX, what was clear brom the beginning. 

Im very happy to get 3840x2160@60hz. Thank you so much!

Maybe it's a bit off topic, but I have one last thing. I can boot with nearly any SMBIOS. But MacPro6,2 wont work. Black Screen and Reboot at the Point the LogIn Screen should be shown. If you have a hint what the problem might is, it would be great.

**Comment by Xenobreaker** [comment]
Yeah, the drivers were loaded, I'm not sure though if it might be a hardware or driver issue cause I can't get my GTX 670 to work at 4k 60hz in Windows either using the Club 3D DisplayPort to HDMI 2.0 adapter. No problems with the GTX 970 though.

**Comment by androcci** [comment]
@Xenobreaker it works for me as simple as downloading windows driver for windows. and then get driver from the nvidia panel. and @Floris497 did you updated the patch?? my gtx 970 is coming in 3 days!!:D

**Comment by Floris497** [comment]
I'll update it later, I currently have no computer near me..it will be done within those 3 days :)

**Comment by androcci** [comment]
@Floris497 floris help! i installed my gtx 970 and i got the option to turn 4k into 60Hz NTSC right out of the box. but when i do the screen went black!! so i switched to my 1080p tv so it can work again. what do you think is the problem here? do you think it was the patch we did before?? thanks!!

**Comment by Floris497** [comment]
Yep.. I think that it works because of the patch on IOKit that is already done, how did those 3 days pass so quickly? What was the exact model of the display?

**Comment by androcci** [comment]
lg smart tv. actually I'm selling it on eBay because I'm the screen is too big for my dorm. here is the link. http://www.ebay.com/itm/LG-40-LED-TV-Ultra-HD-Smart-TV-4K-Wi-Fi-Built-In-40UB8200-LOCAL-PICK-UP-/121872875252?

**Comment by Floris497** [comment]
Looks like the model should support that resolution and framerate just fine.. :/ try using that software see if some framerates around 60Hz work
